flake.nix

```diff
@@ -0,0 +1,152 @@
+{
+  description = "plutus-simple-model";
+
+  inputs = {
+    haskell-nix.url = "github:L-as/haskell.nix/ac825b91c202947ec59b1a477003564cc018fcec";
+    haskell-nix.inputs.nixpkgs.follows = "haskell-nix/nixpkgs-unstable";
+
+    nixpkgs.follows = "haskell-nix/nixpkgs-2105";
+
+    iohk-nix.url = "github:input-output-hk/iohk-nix";
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+
+    # all inputs below here are for pinning with haskell.nix
+    cardano-addresses = {
+      url =
+        "github:input-output-hk/cardano-addresses/d2f86caa085402a953920c6714a0de6a50b655ec";
+      flake = false;
+    };
+    cardano-base = {
+      url =
+        "github:input-output-hk/cardano-base/654f5b7c76f7cc57900b4ddc664a82fc3b925fb0";
+      flake = false;
+    };
+    cardano-crypto = {
+      url =
+        "github:input-output-hk/cardano-crypto/f73079303f663e028288f9f4a9e08bcca39a923e";
+      flake = false;
+    };
+    cardano-ledger = {
+      url =
+        "github:input-output-hk/cardano-ledger/bf008ce028751cae9fb0b53c3bef20f07c06e333";
+      flake = false;
+    };
+    cardano-node = {
+      url =
+        "github:input-output-hk/cardano-node/4f65fb9a27aa7e3a1873ab4211e412af780a3648";
+      flake = false;
+    };
+    cardano-prelude = {
+      url =
+        "github:input-output-hk/cardano-prelude/bb4ed71ba8e587f672d06edf9d2e376f4b055555";
+      flake = false;
+    };
+    cardano-wallet = {
+      url =
+        "github:j-mueller/cardano-wallet/760140e238a5fbca61d1b286d7a80ece058dc729";
+      flake = false;
+    };
+    flat = {
+      url =
+        "github:input-output-hk/flat/ee59880f47ab835dbd73bea0847dab7869fc20d8";
+      flake = false;
+    };
+    goblins = {
+      url =
+        "github:input-output-hk/goblins/cde90a2b27f79187ca8310b6549331e59595e7ba";
+      flake = false;
+    };
+    iohk-monitoring-framework = {
+      url =
+        "github:input-output-hk/iohk-monitoring-framework/46f994e216a1f8b36fe4669b47b2a7011b0e153c";
+      flake = false;
+    };
+    optparse-applicative = {
+      url =
+        "github:input-output-hk/optparse-applicative/7497a29cb998721a9068d5725d49461f2bba0e7a";
+      flake = false;
+    };
+    ouroboros-network = {
+      url =
+        "github:input-output-hk/ouroboros-network/d613de3d872ec8b4a5da0c98afb443f322dc4dab";
+      flake = false;
+    };
+    plutus = {
+      url =
+        "github:input-output-hk/plutus/65bad0fd53e432974c3c203b1b1999161b6c2dce";
+      flake = false;
+    };
+    plutus-apps = {
```

Comment:
Since you're updating all the dependencies anyway, you should use the latest plutus-apps. See https://github.com/mlabs-haskell/plutip/pull/24 for inspiration.